driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -1298,7 +1298,7 @@ static __always_inline void auxmap__store_fdlist_param(struct auxiliary_map *aux
 	push__param_len(auxmap->data, &auxmap->lengths_pos, sizeof(uint16_t) + (num_pairs * (sizeof(int64_t) + sizeof(int16_t))));
 }
 
-static __always_inline void apply_dynamic_snaplen(struct pt_regs *regs, uint16_t *snaplen, bool only_port_range)
+static __always_inline void apply_dynamic_snaplen(struct pt_regs *regs, uint16_t *snaplen, bool only_port_range, enum syscall_nr syscall_number)
```

Comment:
uhm here I think we could use the same approach of `auxmap__store_socktuple_param()` where we pass a pointer to `struct sockaddr *` and then if the port is `0` we use this userspace struct to obtain it 

```c
/* Kernel doesn't always fill sk->__sk_common in sendto and sendmsg syscalls (as in the case of an UDP connection).
 * We fallback to the address from userspace when the kernel-provided address is NULL */ 
if (port_remote == 0 && usrsockaddr != NULL)
{
	struct sockaddr_in usrsockaddr_in = {};
	bpf_probe_read_user(&usrsockaddr_in, bpf_core_type_size(struct sockaddr_in), (void *)usrsockaddr);
	ipv4_remote = usrsockaddr_in.sin_addr.s_addr;
	port_remote = usrsockaddr_in.sin_port;
}
```

I think that this kind of approach could simplify the solution without extra enums or duplicated code

Reply:
On board with this change. Should we also pass the `socket_fd` to avoid any double extraction?